### PR TITLE
[Feature] Add compaction grafana metric

### PR DESCRIPTION
**Pull request by choubenson**
### Orignal Metric
Adapt the original metrics to the new compaction framework：

1. deltas total (rowsets/s)
- base_compaction_deltas_total
- cumulative_compaction_deltas_total
- update_compaction_deltas_total

2. deltas total (bytes/s)
- base_compaction_bytes_total
- cumulative_compaction_bytes_total
- update_compaction_bytes_total

3. total compaction requests
- base_compaction_request_total
- cumulative_compaction_request_total
- update_compaction_request_total

4. failed compaction requests
- base_compaction_request_failed
- cumulative_compaction_request_failed
- update_compaction_request_failed

### New Metric
Add new compaction metric:

1. waiting task num
- wait_base_compaction_task_num
- wait_cumulative_compaction_task_num

2. running task num
- running_base_compaction_task_num
- running_cumulative_compaction_task_num
- running_update_compaction_task_num

3. task cost time
- base_compaction_task_cost_time_ms
- cumulative_compaction_task_cost_time_ms
- update_compaction_task_cost_time_ns

4. task rate
- base_compaction_task_byte_per_second
- cumulative_compaction_task_byte_per_second
- update_compaction_task_byte_per_second

The effect is as following:
image
image

## What type of PR is this:

- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [x] Yes, this PR will result in a change in behavior.
- [ ] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [x] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [ ] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 42 / 61 (68.85%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/storage/storage_engine.cpp|0|1|00.00%|[958]|
|:large_blue_circle:|src/http/action/compaction_action.cpp|0|8|00.00%|[120, 121, 123, 125, 146, 147, 148, 151]|
|:large_blue_circle:|src/storage/compaction_manager.cpp|7|16|43.75%|[149, 150, 151, 183, 184, 285, 286, 287, 394]|
|:large_blue_circle:|src/storage/cumulative_compaction.cpp|6|7|85.71%|[37]|
|:large_blue_circle:|src/storage/rowset_merger.cpp|5|5|100.00%|[]|
|:large_blue_circle:|src/storage/base_compaction.cpp|8|8|100.00%|[]|
|:large_blue_circle:|src/util/starrocks_metrics.cpp|10|10|100.00%|[]|
|:large_blue_circle:|src/storage/compaction_task.cpp|6|6|100.00%|[]|

File: be/src/http/action/compaction_action.cpp [hunk 1]
```diff
@@ -138,7 +142,13 @@ Status CompactionAction::do_compaction(uint64_t tablet_id, const string& compact
             if (rowset_ids.empty()) {
                 return Status::InvalidArgument(fmt::format("empty argument. rowset_ids:{}", rowset_ids_string));
             }
-            RETURN_IF_ERROR(tablet->updates()->compaction(mem_tracker, rowset_ids));
+            res = tablet->updates()->compaction(mem_tracker, rowset_ids);
+        }
+        if (!res.ok()) {
+            StarRocksMetrics::instance()->update_compaction_request_failed.increment(1);
+            LOG(WARNING) << "failed to perform update compaction. res=" << res.to_string()
```
Comment (chaoyli) [inline be/src/http/action/compaction_action.cpp:149]: `res.get_error_msg()` instead of `res.to_string()`

Comment (choubenson) [reply be/src/http/action/compaction_action.cpp]: Resolved.

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 60 / 78 (76.92%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/storage/storage_engine.cpp|0|2|00.00%|[958, 966]|
|:large_blue_circle:|src/http/action/compaction_action.cpp|0|11|00.00%|[120, 121, 122, 123, 125, 145, 147, 148, 149, 150, 151]|
|:large_blue_circle:|src/storage/compaction_task.cpp|7|9|77.78%|[188, 191]|
|:large_blue_circle:|src/storage/compaction_manager.cpp|15|18|83.33%|[183, 184, 186]|
|:large_blue_circle:|src/storage/rowset_merger.cpp|4|4|100.00%|[]|
|:large_blue_circle:|src/storage/cumulative_compaction.cpp|9|9|100.00%|[]|
|:large_blue_circle:|src/storage/base_compaction.cpp|12|12|100.00%|[]|
|:large_blue_circle:|src/util/starrocks_metrics.cpp|13|13|100.00%|[]|

**Comment by LiShuMing** [comment]
@chaoyli we may add the grafana template file into starrocks project, so we can update the grafana template in the later pr.

**Review by chaoyli — APPROVED** [review]

**Review by meegoo — APPROVED** [review]

**Comment by meegoo** [comment]
@mergify backport branch-3.2 branch-3.1 branch-3.0 branch-2.5

**Pull request merged**